### ISSUE TITLE
Is it appropriate to apply the view function in char_emb_layer？

### ISSUE THREAD

**Issue by mottled233** [galsang/BiDAF-pytorch#17]
Hi. I saw you applied the view function to change the data's shape from (batch, seq_len, word_len, char_dim) to (batch * seq_len, char_dim, word_len) in BiDAF-pytorch/model/model.py#L83.
But shouldn't this be a transpose operation?

**Comment by galsang** [comment]
Yeah, it seems weird for me, too.
To clarify, it looks more desirable to apply a transpose function first, then the original view function.
Could you try on this modification and make a pull request if there is any difference in results?
Thanks.

**Comment by mottled233** [comment]
Due to my lack of GPU resources, I could not run the entire process for verification. But after a few epochs, I found that the loss of the modified model dropped a little faster (but not particularly noticeably, may be due to the robustness of the neural network).
Anyway, I have made a pull request.
By the way, what would be the more efficient model architecture that you wrote in this file's TODO?

**Comment by galsang** [comment]
Thanks for your pull request, and it is merged.
